force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -1,136 +1,188 @@
+    Map<String, Object> jsonMap = (Map<String, Object>) JSON.deserializeUntyped(
+      jsonResponse
+    );
+    Map<String, Object> addrMap = (Map<String, Object>) jsonMap.get('address');
+    Contact con;
+    // Create a new contact from the JSON response.
+    // Set the contact fields from the JSON response.
+    con = new Contact(
+      Firstname = (String) jsonMap.get('firstName'),
+      LastName = (String) jsonMap.get('lastName'),
+      Email = (String) jsonMap.get('email'),
+      Phone = (String) jsonMap.get('phone'),
+      Birthdate = Date.valueOf((String) jsonMap.get('birthDate')),
+      // Deserialize the address from the JSON response.
+      // Set the address fields on the contact.
+      MailingStreet = (String) addrMap.get('address'),
+      MailingCity = (String) addrMap.get('city'),
+      MailingState = (String) addrMap.get('state'),
+      MailingPostalCode = (String) addrMap.get('postalCode'),
+      MailingCountry = (String) addrMap.get('country')
+    );
+    return con;
+  }
+
+  /*
+   * Question 3
+   * Sends Contact information to an external system https://dummyjson.com/users/add ('DummyJsonUser') to create or update a user based on the provided Salesforce Contact ID.
+   * The method prepares a JSON payload with the contact's information and sends it to the 'DummyJsonUser' add endpoint.
+   * If the callout is successful (HTTP status code between 200 and 299), the method updates the contact's 'DummyJSON_Last_Updated__c' field in Salesforce to the current time.
+   * Salesforce Contact should be used to generate the JSON payload
+   * Fields: Id, FirstName, LastName, Email, Phone
+   * This callout method should work in triggers
+   *
+   * Documentation: https://dummyjson.com/docs/users
+   *
+   * @param contactId The Salesforce Contact ID used to generate the JSON payload for the external system.
+   */
+ @future(callout=true)
+  public static void postCreateDummyJSONUser(String contactId) {
+    Contact con = [SELECT Id, DummyJSON_Id__c, DummyJSON_Last_Updated__c FROM Contact WHERE Id =: contactId WITH 
+    USER_MODE LIMIT 1];
+    // Create HTTP request to send.
+    HttpRequest request = new HttpRequest();
+    // Set the endpoint URL. Use direct URL or for best practices use Named Credential.
+    request.setEndpoint('callout:DummyJsonUsers' + '/add');
+    // Set the HTTP method to POST.
+    request.setMethod('POST');
+    request.setHeader('Content-Type','application/json' );
+    // Set the body using generateDummyJsonUserPayload method.
+    request.setBody(DummyJSONCallout.generateDummyJsonUserPayload(contactId));
+    // Send the HTTP request and get the response.
+      Http http = new Http();
+      HttpResponse response = http.send(request);
+    // If the HTTP response code is successful, update the contact.
+     if (response.getStatusCode() >= 200 && response.getStatusCode() <= 299) {
+      con.DummyJSON_Last_Updated__c = System.Now();
```

Comment:
At this point you could get the contact record, or you can even create a new instance of Contact with the contactId and the DummyJSON_Last_Updated__c and do an update.